File: Likes-Lookup/liking_users.py
import requests
import os
import json

# To set your enviornment variables in your terminal run the following line:
# export 'BEARER_TOKEN'='<your_bearer_token>'


def auth():
    return os.environ.get("BEARER_TOKEN")


def create_url():
    # User fields are adjustable, options include:
    # created_at, description, entities, id, location, name,
    # pinned_tweet_id, profile_image_url, protected,
    # public_metrics, url, username, verified, and withheld
    user_fields = "user.fields=created_at,description"
    # You can replace the ID given with the Tweet ID you wish to like.
    # You can find an ID by using the Tweet lookup endpoint
    id = "1354143047324299264"
    # You can adjust ids to include a single Tweets.
    # Or you can add to up to 100 comma-separated IDs
    url = "https://api.twitter.com/2/tweets/{}/liking_users".format(id)
    return url, user_fields


def create_headers(bearer_token):
    headers = {"Authorization": "Bearer {}".format(bearer_token)}
    return headers


def connect_to_endpoint(url, headers, user_fields):
    response = requests.request("GET", url, headers=headers, params=user_fields)
    print(response.status_code)
    if response.status_code != 200:
        raise Exception(
            "Request returned an error: {} {}".format(
                response.status_code, response.text
            )
        )
    return response.json()


def main():
    bearer_token = auth()
    url, tweet_fields = create_url()
    headers = create_headers(bearer_token)
    json_response = connect_to_endpoint(url, headers, tweet_fields)
    print(json.dumps(json_response, indent=4, sort_keys=True))


if __name__ == "__main__":
    main()

File: Likes-Lookup/liking_users.rb
# This script uses your bearer token to authenticate and retrieve the specified User objects (by ID)

require 'json'
require 'typhoeus'

# The code below sets the bearer token from your environment variables
# To set environment variables on Mac OS X, run the export command below from the terminal:
# export BEARER_TOKEN='YOUR-TOKEN'
bearer_token = ENV["BEARER_TOKEN"]


# You can replace the ID given with the Tweet ID you wish to like.
# You can find an ID by using the Tweet lookup endpoint
id = "1354143047324299264"

url = "https://api.twitter.com/2/tweets/#{id}/liking_users"

params = {
  # User fields are adjustable, options include:
  # created_at, description, entities, id, location, name,
  # pinned_tweet_id, profile_image_url, protected,
  # public_metrics, url, username, verified, and withheld
  "user.fields": "created_at,description",
}

def liking_users(url, bearer_token, params)
  options = {
    method: 'get',
    headers: {
      "User-Agent": "v2LikingUsersRuby",
      "Authorization": "Bearer #{bearer_token}"
    },
    params: params
  }

  request = Typhoeus::Request.new(url, options)
  response = request.run

  return response
end

response = liking_users(url, bearer_token, params)
puts response.code, JSON.pretty_generate(JSON.parse(response.body))
